src/commonMain/kotlin/fr/acinq/lightning/wire/LightningMessages.kt

```diff
@@ -1747,6 +1751,59 @@ data class PhoenixAndroidLegacyInfo(
     }
 }
 
+/**
+ * A message to request a BIP-353's compliant DNS address from our peer. The peer may not respond, e.g. if there are no channels.
+ *
+ * @param languageSubtag IETF BCP 47 language tag (en, fr, de, es, ...) to indicate preference for the words that make up the address
+ */
+data class DNSAddressRequest(override val chainHash: BlockHash, val offer: OfferTypes.Offer, val languageSubtag: String) : LightningMessage, HasChainHash {
+
+    override val type: Long get() = DNSAddressRequest.type
+
+    override fun write(out: Output) {
+        LightningCodecs.writeBytes(chainHash.value, out)
+        val serializedOffer = OfferTypes.Offer.tlvSerializer.write(offer.records)
+        LightningCodecs.writeU16(serializedOffer.size, out)
+        LightningCodecs.writeBytes(serializedOffer, out)
+        LightningCodecs.writeU16(languageSubtag.length, out)
+        LightningCodecs.writeBytes(languageSubtag.toByteArray(charset = Charsets.UTF_8), out)
+    }
+
+    companion object : LightningMessageReader<DNSAddressRequest> {
+        const val type: Long = 35025
+
+        override fun read(input: Input): DNSAddressRequest {
+            return DNSAddressRequest(
+                chainHash = BlockHash(LightningCodecs.bytes(input, 32)),
+                offer = OfferTypes.Offer.decode(LightningCodecs.bytes(input, LightningCodecs.u16(input)).decodeToString()).get(),
```

Comment:
This won't work, it's still using the string encoding! I find that it's *always* useful to add a basic encode/decode unit test in `LightningCodecTestsCommon`, that's a good safety net and makes it easier to implement the same message in eclair.

Reply:
Damn! I already have an integration test with eclair, but this isn't in the code path :-/

Reply:
Fixed in d1daf8de.